livekit-plugins/livekit-plugins-minimax/livekit/plugins/minimax/tts.py

```diff
@@ -0,0 +1,506 @@
+# livekit/plugins/minimax/tts.py (WebSocket version)
+
+# Copyright 2023 LiveKit, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+import asyncio, base64, json, os, weakref, ssl
+from dataclasses import dataclass, replace
+
+from typing import List, Dict, Any
+import websockets
+# <--- Using websockets for WebSocket connections
+from livekit.agents import (
+    APIConnectionError, APIStatusError, APITimeoutError, tts, utils, tokenize,
+    APIConnectOptions, DEFAULT_API_CONNECT_OPTIONS
+)
+# Removed http_context as we're using WebSocket connections now
+from livekit.agents.voice.io import TimedString
+
+from .log import logger
+from .models import (
+    SUPPORTED_VOICES, TTSDefaultBitRates, TTSBitRates, TTSSampleRates, TTSDefaultSampleRates, TTSDefaultVoiceId,
+    TTSModels, TTSVoices, TTSSubtitleType, TTSDefaultEmotion, TTSEmotion, TTSLanguages, TTSDefaultLanguage
+)
+
+MINIMAX_API_BASE_URL = "https://api-uw.minimax.io"
+
```

Comment:
```suggestion
@dataclass
class VoiceSettings:
    speed: float # [0.5, 2]
    vol: float # (0, 10]
    pitch: int # [-12, 12]
    voice_id: TTSVoices
    emotion: TTSEmotions
    
@dataclass
class AudioSettings:
    sample_rate: TTSSampleRates
    bitrate: TTSBitRates
    
@dataclass
class VoiceModifications:
    intensity: int # [-100, 100]
    timbre: int # [-100, 100]
    sound_effecrs: TTSSoundEffects
```
I think it would be good to separate the settings into these dataclasses in the init method and simplify  `_TTSOptions`